include/cell.tcc

```diff
@@ -450,9 +471,18 @@ inline bool mpm::Cell<Tdim>::is_point_in_cell(
   else
     (*xi) = this->transform_real_to_unit_cell(point);
 
-  // Check if the transformed coordinate is within the unit cell (-1, 1)
-  for (unsigned i = 0; i < (*xi).size(); ++i)
-    if ((*xi)(i) < -1. || (*xi)(i) > 1. || std::isnan((*xi)(i))) status = false;
+  // Check if the transformed coordinate is within the unit cell:
+  // between 0 and 1-xi(1-i) if the element is a triangle, and between
+  // -1 and 1 if otherwise
+  if (this->element_->corner_indices().size() == 3) {
```

Comment:
```
double min = -1., max = 1.;
if (this->element_->corner_indices().size() == 3) {
  min = 0;
  max = 1.;
}
for (unsigned i = 0; i < (*xi).size(); ++i)
  if ((*xi)(i) < min || (*xi)(i) > max || std::isnan((*xi)(i)))
    status = false;
```

Reply:
I know this doesn't test the inclined side, but we need a better approach than too many if else conditions

Reply:
Alternatively, we call a function in `element` class to see if a point is in it's unit cell, that would be a lot nicer. I'd go with that. If you wish that could be a separate PR improving this. So leave this section as is, and then let's do a separate PR improving this.